src/RequestHandlerSwooleRunner.php

```diff
@@ -81,43 +167,88 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
-            printf(
-                "%s - %s - %s %s\n",
-                date('Y-m-d H:i:sO', time()),
-                $request->server['remote_addr'],
-                $request->server['request_method'],
-                $request->server['request_uri']
-            );
-            $emit = new SwooleEmitter($response);
-            try {
-                $psr7Request = ($this->serverRequestFactory)($request);
-            } catch (Throwable $e) {
-                // Error in generating the request
-                $this->emitMarshalServerRequestException($emit, $e);
-                return;
-            }
-            $emit->emit($this->handler->handle($psr7Request));
-        });
+        $this->swooleHttpServer->on('start', [$this, 'onStart']);
+        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
         $this->swooleHttpServer->start();
     }
 
+    /**
+     * On start event for swoole http server
+     */
+    public function onStart(SwooleHttpServer $server) : void
+    {
+        printf("Swoole is running at %s:%s\n", $server->host, $server->port);
+    }
+
+    /**
+     * On HTTP request event for swoole http server
+     */
+    public function onRequest(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) {
+        printf(
+            "%s - %s - %s %s\n",
+            date('Y-m-d H:i:sO', time()),
+            $request->server['remote_addr'],
+            $request->server['request_method'],
+            $request->server['request_uri']
+        );
+        if ($this->getStaticResource($request, $response)) {
+            return;
+        }
+        $emit = new SwooleEmitter($response);
+        try {
+            $psr7Request = ($this->serverRequestFactory)($request);
+        } catch (Throwable $e) {
+            // Error in generating the request
+            $this->emitMarshalServerRequestException($emit, $e);
+            return;
+        }
+        $emit->emit($this->handler->handle($psr7Request));
+    }
+
+    /**
+     * Emit marshal server request exception
+     */
     private function emitMarshalServerRequestException(
         EmitterInterface $emitter,
         Throwable $exception
     ) : void {
         $response = ($this->serverRequestErrorResponseGenerator)($exception);
         $emitter->emit($response);
     }
+
+    /**
+     * Get a static resource, if any, and set the swoole HTTP response
+     */
+    private function getStaticResource(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) : bool {
+        $staticFile = $this->docRoot . $request->server['request_uri'];
+        if (! isset($this->cacheTypeFile[$staticFile])) {
+            if (! file_exists($staticFile)) {
+                return false;
+            }
+            $type = pathinfo($staticFile, PATHINFO_EXTENSION);
+            if (! isset($this->allowedStatic[$type])) {
+                return false;
+            }
```

Comment:
I would swap the order of these two conditionals. `pathinfo()` + `isset()` is a cheaper operation than a `file_exists()` call (`pathinfo()` doesn't do any file stat operations), and it doesn't make sense to do a file stat operation if we don't allow the extension anyways.